feature-libs/product-configurator/rulebased/root/interceptor/cpq-access-storage.service.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@angular/core';
+import { AuthService } from '@spartacus/core';
+import { BehaviorSubject, Observable, Subscription, timer } from 'rxjs';
+import {
+  distinctUntilChanged,
+  expand,
+  filter,
+  switchMap,
+  take,
+} from 'rxjs/operators';
+import { CpqAccessData } from './cpq-access-data.models';
+import { CpqAccessLoaderService } from './cpq-access-loader.service';
+
+export abstract class CpqConfiguratorTokenConfig {
+  cpqConfigurator: {
+    /* We should stop using/sending a token shortly before expiration,
+     * to avoid that it is actually expired when evaluated in the target system.
+     * Time given in ms. */
+    tokenExpirationBuffer: number;
+    /* max time in ms to pass until a token is considered expired and refetched,
+     * even if token expiration time is longer */
+    tokenMaxValidity: number;
+    /* min time to pass until a token is refecthed, even if token expiration time is shorter */
+    tokenMinValidity: number;
+  };
+}
+
+export const DefaultCpqConfiguratorTokenConfig: CpqConfiguratorTokenConfig = {
+  cpqConfigurator: {
+    tokenExpirationBuffer: 10000, // ten seconds
+    tokenMaxValidity: 24 * 60 * 60 * 1000, //one day
+    tokenMinValidity: 5000, // five seconds
+  },
+};
+
+const EXPIRED_TOKEN = {
```

Comment:
Let's move it to the `protected` `readonly` field of the class. For extensibility

Reply:
Pick!

Reply:
Done!